### ISSUE TITLE
Failed item after quick moving

### ISSUE THREAD

**Issue by matthieu-vergne** [syncthing/syncthing#5450]
It looks to me as a duplicate of #2211, but since it is blocked for collaborators only I open a new issue.

# Context

I have 2 nodes:
- PC on Ubuntu 16.04, with Syncthing taken from the [latest release](https://github.com/syncthing/syncthing/releases/tag/v1.0.0) link of the website. The version used is [syncthing-linux-amd64-v1.0.0.tar.gz](https://github.com/syncthing/syncthing/releases/download/v1.0.0/syncthing-linux-amd64-v1.0.0.tar.gz). Displayed version on the Web interface: v1.0.0, Linux (64 bit).
- Tablet on Android 7.0, with Syncthing taken from the Play store. Displayed version on the Web interface: v0.14.51, android (ARM)

I have one shared folder, let's call it `share`, for these two nodes. No specific configuration besides a limit of 1GB of free space. Since I am far from reaching it, I assume it is out of topic and we can assume a default configuration of the shared folder.

As of the storage:
- Tablet stores on the internal storage.
- PC stores on an external hard drive in NTFS (to be accessible on Windows too)

Both nodes were continuously running during the whole operation, so they could sync at any time.

# What happened

Yesterday on the tablet, I added 3 files (several hundreds of MB each) to `share`, and then quickly moved 2 of them to `share/myFolder`. Today I checked my PC, and it displays that the folder is up to date (which is true when I look at the content) but on the 2 moved items 1 is listed as failed item. The message displayed is:
> peers who had this file went away, or the file has changed while syncing. will retry later

# What you expected to happen instead

I would expect Syncthing to be aware of file moving. So rather than considering the `share` and `share/myFolder` files as different, it should consider them as the same, and thus don't expect to have the `share` one when it already has the `share/myFolder` one.

In other words, although the error might happen during the state transition, now it should have figured it was a move and the error should have disappeared already.

# Any steps to reproduce the problem

I assume these are the steps:
- Create a shared folder between 2 nodes A & B
- Create a subfolder
- Wait for both nodes to be sync
- Add a file F in A
- Wait for B to see there is F
- Move F to subfolder before it is fully sync on B (F should be big enough or upload should be limited to ensure it is not too fast)
- Let syncing until B fully retrieves F in the subfolder
- B should still display the error for the missing F in the root folder

**Comment by imsodin** [comment]
I can confirm it, it doesn't disappear after a reload, but it does disappear on a pull (even if unrelated, e.g. creating another file, or just pausing and starting the folder). So the problem is real, but at least just cosmetic. I'll hopefully file a PR with a fix soon.

**Comment by matthieu-vergne** [comment]
Oh, right, pausing and restarting removes it. That is good to know!

I hoped that, if it could not be fixed easily, at least a way to manually remove the failed items would be added, but this way is fine. If it cannot be fixed easily, you may consider adding a message with this workaround on the failed items details.
